### DISCUSSION TITLE
Register Services inside the UseCommunityToolkit call

### DISCUSSION THREAD

**Discussion by pictos** [CommunityToolkit/Maui#1208] [New Feature Discussions]
Today to use the Essentials APIs the user must register the methods manually, as you can see here on our sample app

https://github.com/CommunityToolkit/Maui/blob/642b77849dd3992bfca6cae16f415e7a0ff2541c/samples/CommunityToolkit.Maui.Sample/MauiProgram.cs#L193-L195

I suggest we move this registration inside our lib, that way we register the types, and it will be ready for devs to run. I would say is very rare case where the dev will implement their custom version of those services, and if they want, then they override our registration and they will be good to go.

We should measure if it will have a perf. hit, if so we can use `Lazy<T>` to instantiate the default types.

Also, a friendly reminder that MCT isn't linker friendly, so the linker will run on it just if the user set the linker to be `Link all`. Not sure if you want to make MCT linker safer.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
